### ISSUE TITLE
Support for expand margin, and/or extra spacing from edge of ScrollContainer follow focus

### ISSUE THREAD

**Issue by Giganzo** [godotengine/godot-proposals#10244]
### Describe the project you are working on

User interface/option menu

### Describe the problem or limitation you are having in your project

When using follow focus the expand margin on the focus style for buttons is not included, so it visual looks like the button is not entirely in focus. See image:

![Screenshot_20240715_223902](https://github.com/user-attachments/assets/79e18f97-1b12-4eda-b630-684fd8103a46)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

If there was a toggle for the ScrollContainer to include the expand margin it would look like this:

![Screenshot_20240715_223935](https://github.com/user-attachments/assets/0ca508f0-ca8f-4d3f-a1c9-681f6efeba05)

I made an implementation of this, where I also added support for extra spacing between the focused child and the edge's of the container as a theme setting.

With extra spacing:

![Screenshot_20240715_223954](https://github.com/user-attachments/assets/00151389-0349-45f5-bc75-a0eadf927654)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

My idea is it would look something like this:

![Screenshot_20240718_162005](https://github.com/user-attachments/assets/bce123d0-46f3-4f92-97c2-b8daff9bafff)

![Screenshot_20240718_161922](https://github.com/user-attachments/assets/a7b14c66-9748-4752-bf94-51564699c292)

![image](https://github.com/user-attachments/assets/a4949740-d0db-4934-96af-88f6ceb84255)

This was my first time trying to implement something like this, so there might be a better way to do this. Was a little bit tricky figuring out how to get the expand margin.

https://github.com/Giganzo/godot/tree/scroll-container-follow-focus-settings

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It might be possible to write your own version of this in gdscript and send a signal when child gets focused

### Is there a reason why this should be core and not an add-on in the asset library?

Improving the usability of the ScrollContainer

**Comment by Giganzo** [comment]
Should this include control nodes that doesn't have a focus style? As other control nodes can be set to get focus too (or by using ensure_control_visible), and they might have expand margin set on the StyleBox they use. 

Or what about setting the focus to an StyleBoxEmpty, as some people do  (_I agree with the docs about this not being recommended_), but they might use expand margin on any of the other styles. So how should this be handled or not?

**Comment by MonoGameDev** [comment]
This is exactly the kind of behavior I'm looking for.

I'm using `ensure_control_visible()` to bring controls of interest into view, but always only having them at the very edge of the container impacts user experience in some scenarios.

Has this been discussed any further somewhere?

**Comment by Joy-less** [comment]
I agree this would be very useful for UI elements navigable by arrow keys or joypad.

Currently, `follow_focus` doesn't scroll to the end when focusing the end item, making UI appear to get cut off.

Scrolling scrollbar to leftmost position:
<img width=500 alt="Image" src="https://github.com/user-attachments/assets/10095ae5-3a07-406b-a3f0-66724f0274a4" />

Focusing leftmost element with `follow_focus`:
<img width=500 alt="Image" src="https://github.com/user-attachments/assets/22b704b5-55c5-4e32-b5b0-d28c8b0ecd04" />
